### PR TITLE
"Assignment 2"

### PR DESCRIPTION
**Pull request by Ahmed-Al-Badri**
FullStack, Assignment two.

### CODE REVIEW

File: hw2/01-coins.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/** Exercise 01 - Coins **/
+
+// Add your function here
```
Comment (devamp) [inline hw2/01-coins.js]: Not really important but you can always remove unnecessary comments.

File: hw2/01-coins.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+};
+// Sample test cases
+console.log(calculateChange(4.62));
+// $4.62 ==> 4 dollars, 2 quarters, 1 dime, 2 pennies
+console.log(calculateChange(0.16));
+// $0.16 ==> 1 dime, 1 nickel, 1 penny
+console.log(calculateChange(150.11));
+// $150.11 ==> Error: the number is too large
+
+// Add additional test cases here
```
Comment (devamp) [inline hw2/01-coins.js:46]: Always good to add some more test cases to test your program.

File: hw2/02-node-routing.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// http://localhost:5001/welcome should return a status code 200 with a welcome message of your choice in html format
+
+// http://localhost:5001/redirect should redirect the request to '/redirected' by using 302 as the status code / the redirected page should return a redirected message of your choice
+
+// http://localhost:5001/cache should return 'this resource was cached' in html format and set the cache max age to a day
+
+// http://localhost:5001/cookie should return 'cookies… yummm' in plain text and set 'hello=world' as a cookie
+
+// For other routes, such as http://localhost:5001/other, this exercise should return a status code 404 with '404 - page not found' in html format
+const cache = new NodeCache();
```
Comment (devamp) [inline hw2/02-node-routing.js]: remove unused object

File: hw2/03-express-routing.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+app.get("/cookie", (req, res) => {
+  res.status(200);
+  res.set({ "content-type": "text/html", cookie: "hello=world" });
+  res.send("<div>cookie... yummm</div>");
+});
+app.get("/other", (req, res) => {
+  res.status(404);
+  res.set({ "content-type": "text/html" });
+  res.send("<div>404... page not found</div>");
+});
```
Comment (devamp) [inline hw2/03-express-routing.js]: I don't think `/other` is a required route. "Other" should be any route besides the one required by the assignment.

File: hw2/02-node-routing.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    res.write("<div>cookies... yummm</div>");
+    res.end();
+    return;
+  }
+  if (req.url == "/other") {
+    res.writeHead(401, { "content-type": "text/html" });
+    res.write("<div>404 - page not found</div>");
+    res.end();
+    return;
+  }
```
Comment (devamp) [inline hw2/02-node-routing.js:64]: Don't need `/other` route.

**Review by devamp — CHANGES_REQUESTED** [review]

File: hw2/02-node-routing.js (hunk 3 above)
Comment (Puji97) [reply hw2/02-node-routing.js]: Agreed with Devam on this.

**Comment by Puji97** [comment]
You could add few more test cases. Otherwise everything looks good to me.

File: hw2/01-coins.js (hunk 2 above)
Comment (Puji97) [inline hw2/01-coins.js:46]: The current test cases cover basic functionality, but it would be great to add more cases to handle edge scenarios like: Boundary values (e.g., $0.01, $100)

**Review by Puji97 — APPROVED** [review]

File: hw2/01-coins.js (hunk 2 above)
Comment (Ahmed-Al-Badri) [reply hw2/01-coins.js:46]: Added a couple of more tests.

File: hw2/02-node-routing.js (hunk 5 above)
Comment (Ahmed-Al-Badri) [reply hw2/02-node-routing.js:64]: Yes fixed it for 03 and 04 file, but for 02, it was somewhat needed.

File: hw2/03-express-routing.js (hunk 4 above)
Comment (Ahmed-Al-Badri) [reply hw2/03-express-routing.js]: Fixed it for 04 and 03

File: hw2/02-node-routing.js (hunk 3 above)
Comment (Ahmed-Al-Badri) [reply hw2/02-node-routing.js]: Yes, these variables were removed.

File: hw2/01-coins.js (hunk 2 above)
Comment (Ahmed-Al-Badri) [reply hw2/01-coins.js:46]: Added more cases, thanks

File: hw2/01-coins.js (hunk 1 above)
Comment (Ahmed-Al-Badri) [reply hw2/01-coins.js]: Removed these comments

**Review by Ahmed-Al-Badri — COMMENTED** [review]
Added changes
